node/lib/util/commit.js

```diff
@@ -257,6 +307,17 @@ exports.commit = co.wrap(function *(metaRepo, all, metaStatus, message) {
     const subCommits = {};
     let subsChanged = false;
     const commitSubmodule = co.wrap(function *(name) {
```

Comment:
OK, will do. It will be easier to test if I break it out. Also, you are right about the `subsChanged` section; I think having it calculated like this is a legacy of an earlier version of the algorithm where it was a more complicated to compute than it is now.